File: Array/storingElementsAccordingTo_oddEven.py
# Explanation :- Sorting all even index placed elements in ascending 
#                and Sorting all odd index placed elements in Descending.

def storingOddEven(arr, n):
    even = []
    odd = []
    for i in range(n):
        if i % 2 == 0:
            even.append(arr[i])
        else:
            odd.append(arr[i])
    
    even.sort()
    odd.sort(reverse = True)

    j = 0
    k = 0
    for i in range(0,n):
        if i % 2 ==0:
            arr[i]=even[j]
            j=j+1
        else:
            arr[i]=odd[k]
            k=k+1

    return arr


if __name__ == "__main__":
    n = int(input("\nEnter the Array Size: "))
    arr = list(map(int, input("Enter the Array Elements: ").split()))[:n]
    res = storingOddEven(arr, n)
    print("\nSorted List: ", end = " ")
    for i in range(len(res)):
        print(res[i], end = " ")
    print("\n")

File: Cpp Codes/storingElementsAccordingTo_oddEven.cpp
#include<bits/stdc++.h>

using namespace std;

int *storingOddEven(int *arr, int n)
{
    vector<int> even;
    vector<int> odd;
    
    // for even
    for(int i=0; i<n; i++)
    {
        if(i % 2 == 0)
            even.push_back(arr[i]);
        else
            odd.push_back(arr[i]);
    }
    sort(even.begin(), even.end());
    sort(odd.begin(), odd.end(), greater<int>());

    int j = 0;
    int k = 0;
    for(int i=0; i<n; i++)
    {
        if(i % 2 == 0)
        {
            arr[i] = even[j];
            j++;
        }
        else
        {
            arr[i] = odd[k];
            k++;
        }        
    }
    return arr;
}

int main()
{
    int n;
    cout<<"\nEnter the Array Size: ";
    cin>>n;
    int arr[n];
    cout<<"Enter the Array Elements: ";
    for(int i=0; i<n; i++)
        cin>>arr[i];
    
    int *res;
    res = storingOddEven(arr, n);
    cout<<"\nSorted Array Elements are: ";
    for(int i=0; i<n; i++)
        cout<<res[i]<<" ";
    cout<<"\n\n";
    return 0;

}

